metapool/prep.py

```diff
@@ -0,0 +1,424 @@
+import re
+import os
+import gzip
+import warnings
+import pandas as pd
+
+from glob import glob
+from datetime import datetime
+from string import ascii_letters, digits
+
+# TODO: Ideally these should be imported from Qiita
+PREP_COLUMNS = ['sample_name', 'experiment_design_description',
+                'library_construction_protocol', 'platform', 'run_center',
+                'run_date', 'run_prefix', 'sequencing_meth', 'center_name',
+                'center_project_name', 'instrument_model', 'runid',
+                'sample_plate', 'sample_well', 'i7_index_id', 'index',
+                'i5_index_id', 'index2', 'lane', 'sample_project',
+                'well_description']
+
+# put together by Gail, based on the instruments we know of
+INSTRUMENT_LOOKUP = pd.DataFrame({
+    'A00953': {'machine prefix': 'A', 'Vocab': 'Illumina NovaSeq 6000',
+               'Machine type': 'NovaSeq', 'run_center': 'IGM'},
+    'A00169': {'machine prefix': 'A', 'Vocab': 'Illumina NovaSeq 6000',
+               'Machine type': 'NovaSeq', 'run_center': 'LJI'},
+    'M05314': {'machine prefix': 'M', 'Vocab': 'Illumina MiSeq',
+               'Machine type': 'MiSeq', 'run_center': 'KLM'},
+    'K00180': {'machine prefix': 'K', 'Vocab': 'Illumina HiSeq 4000',
+               'Machine type': 'HiSeq', 'run_center': 'IGM'},
+    'D00611': {'machine prefix': 'D', 'Vocab': 'Illumina HiSeq 2500',
+               'Machine type': 'HiSeq/RR', 'run_center': 'IGM'},
+    'MN01225': {'machine prefix': 'MN', 'Vocab': 'Illumina MiniSeq',
+                'Machine type': 'MiniSeq', 'run_center': 'CMI'}}).T
+
+
+def parse_illumina_run_id(run_id):
+    """Parse a run identifier
+
+    Parameters
+    ----------
+    run_id: str
+        The name of a run
+
+    Returns
+    -------
+    str:
+        When the run happened (YYYY-MM-DD)
+    str:
+        Instrument code
+    """
+
+    # of the form
+    # YYMMDD_machinename_XXXX_FC
+    tokens = run_id.split('_', 1)
+
+    if len(tokens) != 2:
+        raise ValueError('Unrecognized run identifier format "%s". The '
+                         'expected format is YYMMDD_machinename_XXXX_FC.' %
+                         run_id)
+
+    # convert illumina's format to qiita's format
+    run_date = datetime.strptime(tokens[0], '%y%m%d').strftime('%Y-%m-%d')
+
+    return run_date, tokens[1]
+
+
+def sample_sheet_to_dataframe(sheet):
+    """Converts the sample section of a sample sheet into a DataFrame
+
+    Parameters
+    ----------
+    sheet: sample_sheet.SampleSheet
+        Object from where to extract the data.
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame object with the sample information.
+    """
+
+    # used to get the right order every time
+    columns = list(sheet.samples[0].keys())
```

Comment:
What's [0]? Also not sure you need to convert to list ... as long as there are no list indexing, it should be fine.

Reply:
Good catch, that's the first sample.